spring-cloud-generator/src/main/java/com/google/cloud/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -113,7 +113,7 @@ public GapicClass generate(GapicContext context, Service service) {
     Expr thisExpr = ValueExpr.withValue(ThisObjectValue.withType(dynamicTypes.get(className)));
     Transport transport = context.transport();
     boolean hasRestOption =
-        (transport.equals(Transport.GRPC_REST) || transport.equals(Transport.REST))
+        transport.equals(Transport.GRPC_REST)
```

Comment:
Thanks, it makes sense. Can we rename this property to make it clearer? Maybe something like `shouldSupportRestForGrpcClient`?

Reply:
In addition, since this logic is shared between two composers, we may want to extract it to a common place(Util or maybe a AbstractSpringComposer).

Reply:
A quick follow-up on this:
I haven't dig into the generator code on rest-only service, but is it true that as far as for spring code, "rest-only" services are expected to generate exactly the same as "grpc-only" service? Can we add a comment somewhere in the code to document this behavior? I think it's helpful for future maintenance to have a quick link on the code base for background.

Reply:
@blakeli0 Yup, renaming and extracting the logic makes sense. Github did not refresh properly for me and I had missed these latest comments when requesting your re-review; please feel free to ignore!

> "rest-only" services are expected to generate exactly the same as "grpc-only" service?

@zhumin8 Yes, that is the current implementation, which I did not expect at first as well. I can definitely add in a comment to explain this along with the `hasRestOption` refactoring above.

Reply:
Update in [8d7954f](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1849/commits/8d7954fc43498ebeccb4a58be5f1972d97814f0a) with both suggestions above. 

> Can we rename this property to make it clearer? Maybe something like shouldSupportRestForGrpcClient?

@blakeli0 One quick clarification - for this comment, did you mean renaming `hasRestOption` and related variables/utils within the generator code, or renaming the exposed Spring configuration property `useRest`? I made changes for the former since the latter would be a breaking change for users, but just wanted to make sure I'm not misunderstanding. Thanks!

Reply:
Yes, the former.